File: data pendukung/soal1.pl
% Fakta dan Aturan Diagnosa Hama

diagnosa(Hama) :-
    daun_menguning,
    tanaman_layu,
    Hama = 'Hama Wereng'.

diagnosa(Hama) :-
    bercak_hitam,
    tanaman_layu,
    Hama = 'Infeksi Jamur Daun'.

diagnosa(Hama) :-
    daun_berlubang,
    \+ daun_menguning,
    Hama = 'Serangan Ulat'.

diagnosa(Hama) :-
    tanaman_layu,
    \+ daun_menguning,
    \+ bercak_hitam,
    \+ daun_berlubang,
    Hama = 'Penyakit Akar Membusuk'.

diagnosa('Tidak teridentifikasi hama') :-
    \+ (daun_menguning; bercak_hitam; daun_berlubang; tanaman_layu).

% Untuk input manual
tanya(Gejala) :-
    format('Apakah ~w? (y/n): ', [Gejala]),
    read(Jawaban),
    (Jawaban == y -> assertz(Gejala) ; true).

% Program utama
start :-
    retractall(daun_menguning),
    retractall(bercak_hitam),
    retractall(daun_berlubang),
    retractall(tanaman_layu),
    nl,
    write('=== Sistem Diagnosa Hama Tanaman ==='), nl,
    tanya(daun_menguning),
    tanya(bercak_hitam),
    tanya(daun_berlubang),
    tanya(tanaman_layu),
    nl,
    write('Hama yang Terdeteksi:'), nl,
    findall(H, diagnosa(H), Hasil),
    list_hasil(Hasil).

list_hasil([]) :- !.
list_hasil([H|T]) :-
    format('- ~w~n', [H]),
    list_hasil(T).

File: data pendukung/soal1.py
def diagnosa_hama(daun_menguning, bercak_hitam, daun_berlubang, tanaman_layu):
    hasil = []
    
    if daun_menguning and tanaman_layu:
        hasil.append("Hama Wereng")
    if bercak_hitam and tanaman_layu:
        hasil.append("Infeksi Jamur Daun")
    if daun_berlubang and not daun_menguning:
        hasil.append("Serangan Ulat")
    if tanaman_layu and not daun_menguning and not bercak_hitam and not daun_berlubang:
        hasil.append("Penyakit Akar Membusuk")
    
    if not hasil:
        hasil.append("Tidak teridentifikasi hama")
    
    return hasil

def input_gejala(prompt):
    jawaban = input(prompt + " (y/n): ").strip().lower()
    return jawaban == 'y'

# Mulai input user
print("=== Sistem Diagnosa Hama Tanaman ===")
daun_menguning = input_gejala("Apakah daun menguning?")
bercak_hitam = input_gejala("Apakah terdapat bercak hitam?")
daun_berlubang = input_gejala("Apakah daun berlubang?")
tanaman_layu = input_gejala("Apakah tanaman layu?")

# Diagnosa
hasil_diagnosa = diagnosa_hama(daun_menguning, bercak_hitam, daun_berlubang, tanaman_layu)

print("\nHama yang Terdeteksi:")
for h in hasil_diagnosa:
    print("- ", h)
